tests/test_algorithm.py

```diff
@@ -387,6 +444,22 @@ def test_order_methods(self):
             )
             algo.run(self.df)
 
+    def test_order_methods_for_future(self):
+        AlgoClasses = [TestOrderAlgorithm,
+                       TestOrderValueAlgorithm,
+                       TestTargetAlgorithm,
+                       TestOrderPercentAlgorithm,
+                       TestTargetValueAlgorithm]
```

Comment:
This might be a good use-case for `nose.parameterized`.  Search around for `parameterized.expand` to see how it's used.